src/config.cpp

```diff
@@ -546,17 +595,51 @@ class SimulationConfig::Parser
             auto& valueIt = it.value();
             const auto debugStr = fmt::format("report {}", it.key());
             parseMandatory(valueIt, "cells", debugStr, report.cells);
-            parseMandatory(valueIt, "type", debugStr, report.type);
+            parseOptional<Report::Sections>(valueIt,
+                                            "sections",
+                                            report.sections,
+                                            Report::Sections::soma);
+            parseMandatory<Report::Type>(valueIt, "type", debugStr, report.type);
+            parseOptional<Report::Scaling>(valueIt,
+                                           "scaling",
+                                           report.scaling,
+                                           Report::Scaling::area);
+            if (report.sections == Report::Sections::soma) {
+                parseOptional<Report::Compartments>(valueIt,
+                                                    "compartments",
+                                                    report.compartments,
+                                                    Report::Compartments::center);
+            } else {
+                parseOptional<Report::Compartments>(valueIt,
+                                                    "compartments",
+                                                    report.compartments,
+                                                    Report::Compartments::all);
+            }
+            parseMandatory(valueIt, "variable_name", debugStr, report.variableName);
+            parseOptional<std::string>(valueIt, "unit", report.unit, "mV");
             parseMandatory(valueIt, "dt", debugStr, report.dt);
             parseMandatory(valueIt, "start_time", debugStr, report.startTime);
             parseMandatory(valueIt, "end_time", debugStr, report.endTime);
-            parseOptional(valueIt, "file_name", report.fileName);
-            if (report.fileName.empty())
-                report.fileName = it.key() + "_SONATA.h5";
-            else {
-                const auto extension = fs::path(report.fileName).extension().string();
-                if (extension.empty() || extension != ".h5")
-                    report.fileName += ".h5";
+            parseOptional<std::string>(valueIt,
+                                       "file_name",
+                                       report.fileName,
+                                       it.key() + "_SONATA.h5");
+            parseOptional(valueIt, "enabled", report.enabled);
+
+            checkValidEnum(report.sections);
+            checkValidEnum(report.scaling);
+            checkValidEnum(report.compartments);
+
+            // Validate comma separated strings
+            std::regex expr(R"(\w+(\s*,\s*\w+)*)");
```

Comment:
are the capture groups needed?

Reply:
Yes, otherwise it doesn't compile.
https://www.geeksforgeeks.org/raw-string-literal-c/

Reply:
The inner one though: `(\s*,\s*\w+)`?

Reply:
Then single values fail ("variable") and wrong comma separation pass ("variable,")

Reply:
But they're not being captured, correct, it's just to group.  Ie:
`\w+(?:\s*,\s*\w+)*` would work?

Reply:
do you mean this?
`std::regex expr(R"(\w+(?:\s*,\s*\w+)*)");`

Reply:
yeah; I think so